### PR TITLE
hub: Replace card_spaces/merchant_infos with profiles

### PR DESCRIPTION
**Pull request by backspace**
This is mostly a mechanical process of combining the creation of separate models into a single profile. Since the surviving fields on `card_spaces` were all optional and it had a foreign key for merchants, it’s as if it merged into `merchant_infos`.

This preserves the existing endpoints at `/api/card-spaces` and `/api/merchant-infos` so we don’t have to update all clients simultaneously. Then we can add `/api/profiles` and eventually remove those once everything has migrated. I did delete `POST /api/card-spaces` because it’s no longer applicable.

### CODE REVIEW

File: packages/hub/config/structure.sql [hunk 1]
```diff
@@ -1095,6 +1062,26 @@ CREATE TABLE public.prepaid_card_patterns (
...
 ALTER TABLE public.prepaid_card_patterns OWNER TO postgres;
 
+--
+-- Name: profiles; Type: TABLE; Schema: public; Owner: postgres
+--
+
+CREATE TABLE public.profiles (
+    id uuid NOT NULL,
+    name text NOT NULL,
+    slug text NOT NULL,
```
Comment (jurgenwerk) [inline packages/hub/config/structure.sql:1072]: We're querying by slug so would be good to add an index

File: packages/hub/db/migrations/20220728144935996_add-profiles.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    INSERT INTO profiles
+    SELECT
+      m.id, m.name, m.slug, m.color, m.text_color, m.owner_address,
+      c.links, c.profile_image_url, c.profile_description,
+      m.created_at
+    FROM merchant_infos as m
+    LEFT JOIN card_spaces AS c ON m.id = c.merchant_id
+  `);
+
+  await pgm.db.query('DROP TABLE card_spaces');
```
Comment (jurgenwerk) [inline packages/hub/db/migrations/20220728144935996_add-profiles.ts:43]: I'm extra paranoid about dropping tables in production before verifying the migration + new code works ok 🫥

**Review by jurgenwerk — APPROVED** [review]
Looking good!

File: packages/hub/config/structure.sql (hunk 1 above)
Comment (backspace) [reply packages/hub/config/structure.sql:1072]: eep ya, I missed [that](https://github.com/cardstack/cardstack/pull/3120/files#diff-e03f308a75e917163fbf9fe5bc34d68936b918230eea5447cb6f580bf7af8c70L1484), thanks! Included in 1361d9d

File: packages/hub/db/migrations/20220728144935996_add-profiles.ts (hunk 2 above)
Comment (backspace) [reply packages/hub/db/migrations/20220728144935996_add-profiles.ts:45]: yes, it’s scary! I ran this many times locally to refine it. I’ll be careful about preserving everything before running it in production.

**Pull request merged**